### DISCUSSION TITLE
How to link old versions?

### DISCUSSION THREAD

**Discussion by ddnexus** [retypeapp/retype#299] [Ideas]
When used for documentation, it would be nice to have some way to link to old versions of the doc.

Currently github pages does not allow multiple version of the same doc, and that sucks, but retype could bypass that restriction e.g. including the old version(s) in the build. 

Ideally, the config could include the possibility to build from an array of tags present in the repo, and there could be a component that shows a popup menu jumping to the old versions.

**Comment by geoffreymcgill** [comment]
We have versioning as a goal, although it will be while until we can implement. 

A very simple way of archiving copies of your content would be to create the folders `/Archives/v1.0/`, and place a copy of the content you want to version into that `/Archives/v1.0/` folder. 

The when you have another version of the content you want to version, make a `/Archives/v2.0/` folder and copy the content again. 

Name the folders whatever you want. The above `/Archives/v#.#` is just a suggestion.

**Reply by ddnexus** [reply]
Is there any trick to make retype do a sort of chroot to the version dir, so I could just copy the content of the old build and get everything as it was?

**Reply by geoffreymcgill** [reply]
I'm not sure I 100% understand your request, but maybe passing an `--override` of the `input` config to `retype build` would help?
https://retype.com/guides/cli/#basic-config

Something like:
```
retype build --override "{ \"input\": \"./Archives/v1.0/" }"
```

Is that what you mean?

That would also mean copying the entire contents of your project into the `/Archives/v1.0/` folder, including the `retype.yml` file, but that should work and should be okay. 

Hope this helps.

**Comment by ddnexus** [comment]
That's also good to know for the build step. Thank you.

What I meant was: using the full build from an old version and copy it in `/Archives/v#/`, then having some trick e.g. to switch/restart the whole retype site into that dir, maybe injecting a `base` tag in the HTM, so all the old artifacts would replace the current, including menus , categories... everything. 

Not sure that would be possible (probably not), but handling a base tag might solve the problem without reimplementing everything.
